thirdparty/CMakeLists.txt

```diff
@@ -7,24 +7,33 @@ set(YAML_CPP_BUILD_CONTRIB OFF CACHE BOOL "")
 set(YAML_BUILD_SHARED_LIBS OFF CACHE BOOL "")
 set(YAML_CPP_INSTALL OFF CACHE BOOL "")
 set(YAML_CPP_FORMAT_SOURCE OFF CACHE BOOL "")
-FetchContent_Declare(
-    yaml-cpp
-    GIT_REPOSITORY      https://github.com/jbeder/yaml-cpp
-    GIT_TAG             1b50109f7bea60bd382d8ea7befce3d2bd67da5f
-)
 
-FetchContent_MakeAvailable(yaml-cpp)
-loco_thirdparty_target_compile_link_flags(yaml-cpp)
+find_package(yaml-cpp)
+
+IF ( NOT yaml-cpp_FOUND )
+    FetchContent_Declare(
+        yaml-cpp
+        GIT_REPOSITORY      https://github.com/jbeder/yaml-cpp
+        GIT_TAG             1b50109f7bea60bd382d8ea7befce3d2bd67da5f
```

Comment:
All you need to do here is add FIND_PACKAGE_ARGS then you can remove all the other changes

Reply:
I've tried this on a separate branch. It seems to have issues with loco_thirdparty_target_compile_link_flags. 
```
CMake Error at cmake/OpenLocoUtility.cmake:42 (target_compile_options):
  target_compile_options may only set INTERFACE properties on IMPORTED
  targets
Call Stack (most recent call first):
  thirdparty/CMakeLists.txt:18 (loco_thirdparty_target_compile_link_flags)


CMake Error at thirdparty/CMakeLists.txt:32 (set_target_properties):
  set_target_properties Can not find target to add properties to: span-lite
```

I'll try reinstating the last check and see if that means we can reduce it to one.

Reply:
It looks like this needs to stay. I think it is because loco_thirdparty_target_compile_link_flags only needs to be done if we've not been able to find the package.

Reply:
I changed it to have a single wrapped condition around the block rather than having three separate conditions. This cleans it up a bit.

Reply:
Hmm I'm just reading that a lot of the fetchcontent things that would be needed are unfortunately 3.24 and 3.25 features which the ci's wont have.